engine/redis/engine.go

```diff
@@ -82,6 +82,33 @@ func (e *Engine) Publish(namespace, queue string, body []byte, ttlSecond, delayS
 	return job.ID(), err
 }
 
+func (e *Engine) BatchConsume(namespace, queue string, count, ttrSecond, timeoutSecond uint32) (jobs []engine.Job, err error) {
+	jobs = make([]engine.Job, 0)
+	for i := uint32(0); i < count; i++ {
+		job, err := e.Consume(namespace, queue, ttrSecond, 0)
```

Comment:
// `timeout` is 0 to fast check whether there are any jobs in the ready queue,
// if any, we wouldn't be blocking until the new job was published.